pilot/pkg/networking/core/cluster_tls.go

```diff
@@ -324,7 +324,7 @@ func (cb *ClusterBuilder) applyHBONETransportSocketMatches(c *cluster.Cluster, t
 				if tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
 					// If a user sets ISTIO_MUTUAL, then HBONE is replacing it. So we will do HBONE or mTLS, depending on backend support.
 					c.TransportSocketMatches = []*cluster.Cluster_TransportSocketMatch{
-						hboneTransportSocket(ts),
+						hboneTransportSocket(xdsfilters.RawBufferTransportSocket),
```

Comment:
What's the motivation for this change?

Reply:
Before this was putting mTLS over HBONE which is super broken. The HBONe path never activated though (due to the EDS code I had to change) so it wasn't noticed